Geometry/HGCalCommonData/src/HGCalGeomParameters.cc

```diff
@@ -261,13 +261,227 @@ void HGCalGeomParameters::loadGeometryHexagon(const DDFilteredView& _fv,
       layers, trforms, trformUse, copies, copiesInLayers, wafer2copy, wafers, wafertype, cellsf, cellsc, php);
 }
 
-void HGCalGeomParameters::loadGeometryHexagon(const cms::DDFilteredView& fv,
+void HGCalGeomParameters::loadGeometryHexagon(cms::DDFilteredView& fv,
                                               HGCalParameters& php,
                                               const std::string& sdTag1,
                                               const cms::DDCompactView* cpv,
                                               const std::string& sdTag2,
                                               const std::string& sdTag3,
-                                              HGCalGeometryMode::WaferMode mode) {}
+                                              HGCalGeometryMode::WaferMode mode) {
+  bool dodet(true);
+  std::map<int, HGCalGeomParameters::layerParameters> layers;
+  std::vector<HGCalParameters::hgtrform> trforms;
+  std::vector<bool> trformUse;
+
+  while (dodet) {
+    const std::vector<double>& pars = fv.parameters();
+    // Layers first
+    std::vector<int> copy = fv.history().copyNos;
+    int nsiz = (int)(copy.size());
+    int lay = (nsiz > 0) ? copy[nsiz - 1] : 0;
+    int zp = (nsiz > 2) ? copy[nsiz - 3] : -1;
+    if (zp != 1)
+      zp = -1;
+    if (lay == 0) {
+      throw cms::Exception("DDException") << "Funny layer # " << lay << " zp " << zp << " in " << nsiz << " components";
+    } else {
+      if (std::find(php.layer_.begin(), php.layer_.end(), lay) == php.layer_.end())
+        php.layer_.emplace_back(lay);
+      auto itr = layers.find(lay);
+      double zp = HGCalParameters::k_ScaleFromDD4Hep * fv.translation().Z();
+      if (itr == layers.end()) {
+        double rin(0), rout(0);
+        if (fv.isA<dd4hep::Polyhedra>()) {
+          rin = 0.5 * HGCalParameters::k_ScaleFromDD4Hep * (pars[5] + pars[8]);
+          rout = 0.5 * HGCalParameters::k_ScaleFromDD4Hep * (pars[6] + pars[9]);
```

Comment:
We don't want to have dependence on the value and order of these arbitrary shape parameters scattered throughout our code. I need to create a DDPolyHedron shape from which these rin/rout values can be obtained.
Please add a FIXME comment here, and I will fix this issue in a later PR.